lib/fishjam/room/id.ex

```diff
@@ -8,20 +8,17 @@ defmodule Fishjam.Room.ID do
   @doc """
   Based on the Room ID determines to which node it belongs to.
   Returns an error if the node isn't present in the cluster.
-
-  DISCLAIMER:
-    It should be used only with room_ids generated by generate/0, otherwise it can raise.
   """
   @spec determine_node(id()) ::
-          {:ok, node()} | {:error, :invalid_room_id} | {:error, :invalid_node}
+          {:ok, node()} | {:error, :invalid_room_id | :node_not_found}
   def determine_node(room_id) do
     with {:ok, room_id} <- validate_room_id(room_id),
-         node_name <- decode_node_name(room_id),
+         {:ok, node_name} <- decode_node_name(room_id),
          true <- node_present_in_cluster?(node_name) do
-      {:ok, node_name}
+      {:ok, String.to_atom(node_name)}
```

Comment:
`String.to_existing_atom` to prevent creating new atoms. Node name is a registered atom so it shouldn't ever fail.